Car Port: 2022 Honda Civic

sshane opened the pull request:
Moved from: https://github.com/commaai/openpilot/pull/23307

***** Template: Car port *****

**Checklist**
- [x] added to README
- [x] test route added to [test_routes.py](https://github.com/commaai/openpilot/blob/master/selfdrive/test/test_routes.py)
- [x] route with openpilot: f44aa96ace22f34a|2021-12-22--06-22-31
- [x] route with stock system: f44aa96ace22f34a|2021-12-25--21-08-31
- [x] confirm spam resume works
- [x] ~~figure out possible new LKAS_HUD message~~

Requires: 
- [x] https://github.com/commaai/cereal/pull/234 - will make safety param
- [x] https://github.com/commaai/opendbc/pull/620
- [x] https://github.com/commaai/panda/pull/942

Lateral only for now, long should be pretty easy. There is no radar on this car so openpilot can send ACC commands directly to the car.

vanillagorillaa committed:
master 2022 Civic

vanillagorillaa committed:
bump panda

vanillagorillaa committed:
bump

vanillagorillaa committed:
bump cereal

vanillagorillaa committed:
fix

vanillagorillaa committed:
needed

vanillagorillaa committed:
try for now

vanillagorillaa committed:
maybe

vanillagorillaa committed:
revert for now

vanillagorillaa committed:
move to Cam parser

vanillagorillaa committed:
fix

vanillagorillaa committed:
move to cam

vanillagorillaa committed:
need AEB_STATUS too

vanillagorillaa committed:
bump for debug prints

vanillagorillaa committed:
bump opendbc and add cruise_params

vanillagorillaa committed:
bump opendbc and update cruise_params

vanillagorillaa committed:
bump

vanillagorillaa committed:
test route

vanillagorillaa committed:
update ref

vanillagorillaa committed:
Revert "update ref"

This reverts commit 28345dab63d1919865ccb510265222a4cd4252f4.

vanillagorillaa committed:
cleanup

vanillagorillaa committed:
Merge branch 'master' into civic22_master

vanillagorillaa committed:
just to test

vanillagorillaa committed:
bump

vanillagorillaa committed:
revert

vanillagorillaa committed:
need to send val 12 too?

vanillagorillaa committed:
change bus

vanillagorillaa committed:
not needed

vanillagorillaa committed:
update bus

vanillagorillaa committed:
syntax

vanillagorillaa committed:
move this to other bus too

vanillagorillaa committed:
Revert "move this to other bus too"

This reverts commit 770bf4745ee244c8426ac108f44b67777198d0a7.

vanillagorillaa committed:
test new lane line signal

vanillagorillaa committed:
needed too

vanillagorillaa committed:
maybe need both?

vanillagorillaa committed:
Test new LKAS hud message

vanillagorillaa committed:
bump

vanillagorillaa committed:
missing comma

vanillagorillaa committed:
missing

vanillagorillaa committed:
maybe

vanillagorillaa committed:
add frame and idx

vanillagorillaa committed:
add in hud_lanes

vanillagorillaa committed:
switch this too

vanillagorillaa committed:
bump panda

vanillagorillaa committed:
add this

vanillagorillaa committed:
I guess need this too

vanillagorillaa committed:
to match

vanillagorillaa committed:
also

vanillagorillaa committed:
wasnt correct

vanillagorillaa committed:
bump opendbc

vanillagorillaa committed:
bump panda

vanillagorillaa committed:
move to cam parser

vanillagorillaa committed:
missing

vanillagorillaa committed:
add here too

vanillagorillaa committed:
bump

vanillagorillaa committed:
remove from cam parser

vanillagorillaa committed:
bump

vanillagorillaa committed:
back to cam parser

vanillagorillaa committed:
its 5hz

vanillagorillaa committed:
bump for new checksum function

vanillagorillaa committed:
bump for correct frequency

vanillagorillaa committed:
update frame and idx

vanillagorillaa committed:
bump

vanillagorillaa committed:
bump and update

vanillagorillaa committed:
send set me bit

vanillagorillaa committed:
bump

vanillagorillaa committed:
pass these values through

vanillagorillaa committed:
silly atom

vanillagorillaa committed:
ret

vanillagorillaa committed:
fix this

vanillagorillaa committed:
use copy instead

vanillagorillaa committed:
add these too

vanillagorillaa committed:
to check keyerror

vanillagorillaa committed:
switch

vanillagorillaa committed:
bump submodules

vanillagorillaa committed:
Merge branch 'master' of https://github.com/commaai/openpilot into commaai-master

vanillagorillaa committed:
Merge branch 'commaai-master' into civic22_master

vanillagorillaa committed:
send too

vanillagorillaa committed:
proper

kevinharbin committed:
Replace HUD with BOH

vanillagorillaa committed:
Merge pull request #4 from kevinharbin/patch-2

Replace HUD with BOH

vanillagorillaa committed:
add dashed lanes

vanillagorillaa committed:
small fix

vanillagorillaa committed:
clean up

vanillagorillaa committed:
not needed anymore

vanillagorillaa committed:
remove and change

vanillagorillaa committed:
this too

vanillagorillaa committed:
dont always set

vanillagorillaa committed:
remove additional LKAS message

vanillagorillaa committed:
bump

vanillagorillaa committed:
add

vanillagorillaa committed:
to test

vanillagorillaa committed:
add frame

vanillagorillaa committed:
bump

vanillagorillaa committed:
rebase

vanillagorillaa committed:
Merge branch 'master' of https://github.com/commaai/openpilot into commaai-master

vanillagorillaa committed:
Merge branch 'commaai-master' into civic22_master

vanillagorillaa committed:
remove default values

sshane committed:
Merge remote-tracking branch 'upstream/master' into civic22_master

sshane committed:
rename

sshane committed:
clean up some carstate logic

sshane committed:
regenerate docs

sshane committed:
spacing

sshane committed:
simplify more logic

sshane committed:
bump opendbc

sshane committed:
bump opendbc

sshane committed:
Merge remote-tracking branch 'upstream/master' into civic22_master

sshane committed:
only if radarless

sshane committed:
panda at least builds now

sshane committed:
add comment

sshane committed:
Merge remote-tracking branch 'upstream/master' into civic22_master

sshane committed:
bump

sshane committed:
fixes

sshane committed:
bump opendbc

sshane committed:
bump opendbc

fix for new DBC

sshane committed:
bump opendbc

sshane committed:
bump opendbc

sshane committed:
carstate: fix bus, parser signals

sshane committed:
Set safety param

sshane commented:
`ACC_HUD` is on bus 2
`ACC_CONTROL` is on bus 0
`POWERTRAIN_DATA` and all other signals in `get_can_signals` (pt) are on bus 0

sshane committed:
pt bus is 0, not 1

sshane committed:
Fix SCM_BUTTONS and bump panda and opendbc

sshane committed:
fixes for ACC_CONTROL

sshane committed:
bump opendbc

sshane committed:
bump opendbc

sshane committed:
convert from MPH on HONDA_BOSCH_RADARLESS

move is_metric

sshane committed:
make sure we don't disable if radarless

sshane committed:
Merge remote-tracking branch 'upstream/master' into civic22_master

sshane committed:
don't show incorrect harness on website

don't show incorrect harness on website

sshane committed:
bump panda

kevinharbin commented:
What's the criteria for "Steer to 0"? I think the 22 civic does that.

adeebshihadeh commented:
Ignition issue: #24680

> Describe the bug
> The car went onroad/recording with the car off, parked, and no one around.
> There were 29 0 min routes starting at 11:49am then a 48 minute route starting at 11:53 that ended when I got back to the car and unplugged the OBD power.
> 
> I don't know how to reproduce the bug.
> 
> Provide a route where the issue occurs
> f44aa96ace22f34a|2022-05-29--11-49-34--0

sshane committed:
Merge remote-tracking branch 'upstream/master' into civic22_master

sshane committed:
remove/update comments

sshane committed:
bump panda

sshane committed:
Merge remote-tracking branch 'upstream/master' into civic22_master

sshane committed:
Fix harnesses

sshane committed:
Merge remote-tracking branch 'upstream/master' into civic22_master

sshane committed:
one line check

sshane committed:
bump opendbc

sshane committed:
Merge remote-tracking branch 'upstream/master' into civic22_master

sshane committed:
remove this

sshane committed:
Some carstate cleanup

We removed STANDSTILL->WHEELS_MOVING

we don't use CRUISE_PARAMS

add back

add back

sshane committed:
more cleanup

sshane committed:
update docs

sshane committed:
marketing says it has TJA and ACC with low speed follow

sshane committed:
send buttons on bus 0

bump panda

sshane committed:
Merge remote-tracking branch 'upstream/master' into civic22_master

sshane committed:
comment

sshane committed:
camera needs to see buttons on bus 2

comment

sshane commented:
This user still seems to be having this problem, device turns on and off multiple times for just a few seconds. None of the logs I checked had started or any ignition signals high, so it must be for just a second and it takes a while for loggerd to start

![Screenshot from 2022-06-14 11-26-45](https://user-images.githubusercontent.com/25857203/173662580-f89eafc4-ab62-4184-9ac1-055911767edc.png)

sshane committed:
Merge remote-tracking branch 'upstream/master' into civic22_master

sshane committed:
Merge remote-tracking branch 'upstream/master' into civic22_master

sshane committed:
Merge remote-tracking branch 'upstream/master' into civic22_master

sshane committed:
bump panda

adeebshihadeh commented on selfdrive/car/honda/values.py:
```diff
@@ -1392,6 +1394,40 @@ class HondaCarInfo(CarInfo):
+  CAR.CIVIC_2022: {
+    (Ecu.eps, 0x18DA30F1, None): [
+      b'39990-T39-A130\x00\x00',
+      b'39990-T43-J020\x00\x00',  # hatch
```
these comments aren’t necessary and probably won’t be maintained

adeebshihadeh commented on selfdrive/car/honda/values.py:
```diff
@@ -75,6 +75,7 @@ class CAR:
   CIVIC = "HONDA CIVIC 2016"
   CIVIC_BOSCH = "HONDA CIVIC (BOSCH) 2019"
   CIVIC_BOSCH_DIESEL = "HONDA CIVIC SEDAN 1.6 DIESEL 2019"
+  CIVIC_2022 = "HONDA CIVIC 2022"
```
we should probably start moving to using generations like the VW port

sshane replied:
I'm fine with making a migration later on if it turns out all new Hondas (/Civics) are like this.

adeebshihadeh commented on selfdrive/car/honda/values.py:
```diff
@@ -1429,5 +1466,6 @@ class HondaCarInfo(CarInfo):
-               CAR.CRV_HYBRID, CAR.INSIGHT, CAR.ACURA_RDX_3G, CAR.HONDA_E}
-HONDA_BOSCH_ALT_BRAKE_SIGNAL = {CAR.ACCORD, CAR.CRV_5G, CAR.ACURA_RDX_3G}
+               CAR.CRV_HYBRID, CAR.INSIGHT, CAR.ACURA_RDX_3G, CAR.HONDA_E, CAR.CIVIC_2022}
+HONDA_BOSCH_ALT_BRAKE_SIGNAL = {CAR.ACCORD, CAR.CRV_5G, CAR.ACURA_RDX_3G}  # no radarless car should have this signal
```
is this necessarily true or just happens to be true now?

sshane replied:
We assert this in safety right now to avoid accidentally adding a new platform to the list where we haven't seen a car yet with the other brake signal. I'll remove comment though

sshane committed:
add to releases

sshane committed:
Merge remote-tracking branch 'upstream/master' into civic22_master

sshane committed:
remove comments

sshane committed:
comment

sshane committed:
we don't use stock hud yet
